paasta_tools/kubernetes_tools.py

```diff
@@ -4354,6 +4425,20 @@ def get_kubernetes_secret_env_variables(
     return decrypted_secrets
 
 
+def create_temp_exec_token(kube_client: KubeClient, namespace: str, user: str):
+    """Create a short lived token for exec"""
+    audience = "remote-run-" + user
+    service_account = "default"
+    token_spec = V1TokenRequestSpec(
+        expiration_seconds=600, audiences=[audience]  # minimum allowed by k8s
```

Comment:
The audience will need to be the kubernetes one, otherwise it won't be usable for doing execs. You should be able to just leave it unset to achieve that.